### PR TITLE
TPO page: project card links to project page, donate button opens donation dialog

### LINKED ISSUE #153: To Do List for 7th Sept
**Donations -** 
- [x] Some projects don't have gateways I believe, so donation is stuck on "Seeding your donation" page

- [x] Country search by name and don't show the code
- [x] Hide PayPal
- [x] Adding commas in the price
- [x] SCA needs to be done
- [x] Change share button design
- [x] Recipient name instead of First Name, last name in gift
- [x] Going back from payment page clears the country selection.
- [x] Country sometimes goes as Null
- [x] Thank you page – commas in numbers trees and currency.

**Zoom 1 -**
- [x] Scroll on Tablet/mobile stops working suddenly
- [x] Projects Scroll is glitchy.
- [x] In popup donate takes to the single project, this shouldn't happen
- [x] Fly to single project, fix onTransitionEnd callback

**TPO Page -**
- [x] On tpo page the card needs to be responsive
- [ ] In tpo page clicking on project doesn't work and also not donate
- [x] The tpo URL is to be removed and t has to be used

**Designs -**
- [x] Body need to use the primary font family
- [x] Country popup of footer needs design change
- [x] Footer should be larger on the mobile
- [x] Responsive designs for projects and opened project

**Tenants -**
- [x] For salesforce page title should be home instead of leaders and use world icon
- [x] Designs to be updated for salesforce with their designs

This is a feature and not a bug - So that the undefined values aren't used
- [x] Custom trees option doesn't get selected until I type something.

### PR DESCRIPTION
**Pull request by sarveshpro**
Fixes #153 

Changes in this pull request:
- project card link to project page
- donate button open donation modal

### CODE REVIEW

**Comment by sarveshpro** [comment]
There's one error I am getting couldn't resolve. when clicked on donate button on tpo page, it opens the donation modal but the app crashed then. error is caused in getFormatedCurrency.ts

**Comment by norbertschuler** [comment]
I rebased this patch onto the develop branch and fixed some errors, but still not working for me at http://localhost:3000/t/plant-for-the-planet as an API call fails with 404.

File: src/features/public/PublicUserProfile/components/ProjectSnippet.tsx [hunk 1]
```diff
@@ -12,79 +19,136 @@ interface Props {
...
+  }, []);
+  const projectDetails = {
+    id: project.id,
+    name: project.name,
+    currency: project.currency,
+    country: project.country,
+    treeCost: project.treeCost,
+    taxDeductionCountries: taxDeductionCountriesArray,
+    tpo: project.tpoData,
+  };
```
Comment (norbertschuler) [inline src/features/public/PublicUserProfile/components/ProjectSnippet.tsx:62]: projectDetails meanwhile are filled with `project.properties` as returned by https://app-development.plant-for-the-planet.org/app/projects?_scope=map&currency=EUR - this data structure does not work anymore, so we will have change this structure here for the DonationsPopup component to work:

Bildschirmfoto 2020-10-29 um 18 10 54

Comment (norbertschuler) [reply src/features/public/PublicUserProfile/components/ProjectSnippet.tsx:62]: At best we would be able to load these project properties for a single project before opening the donation popup? Is there an API call for this?

Comment (norbertschuler) [reply src/features/public/PublicUserProfile/components/ProjectSnippet.tsx]: I fixed it by doing single API calls for every project. Working but at the price of several API calls.

**Comment by norbertschuler** [comment]
@sarveshpro please check my changes.

**Comment by norbertschuler** [comment]
Can be closed in favor of the solution integrated in #341.

**Comment by norbertschuler** [comment]
Closing in favor of https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/341

**Pull request closed without merging**